src/components/Badge/__tests__/badge-tests.jest.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import "@testing-library/jest-dom";
+import { render, cleanup } from "@testing-library/react";
+import Badge, { BadgeProps } from "../Badge";
+import { ComponentDefaultTestId } from "../../../tests/constants";
+import Indicator from "../../Indicator/Indicator";
+import { IndicatorColor } from "../../Indicator/IndicatorConstants";
+import Counter from "../../Counter/Counter";
+
+const DEFAULT_CHILD = "Child Text";
+
+const renderComponent = (props: BadgeProps = {}, children: React.ReactNode = DEFAULT_CHILD) => {
+  return render(<Badge {...props}>{children}</Badge>);
+};
+
+describe("Badge", () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  it("should render correctly", () => {
+    const { getByTestId } = renderComponent();
+    expect(getByTestId(ComponentDefaultTestId.BADGE)).toBeInTheDocument();
+  });
+
+  it("should be shown", () => {
+    const { getByTestId } = renderComponent();
+    expect(getByTestId(ComponentDefaultTestId.BADGE)).toHaveClass("show");
+  });
+
+  it("should render when has no children", () => {
+    const { getByTestId, queryByText } = renderComponent({}, null);
+    const badgeElement = getByTestId(ComponentDefaultTestId.BADGE);
```

Comment:
If we have proper snapshots there is no need in this test + children should be mandatory